e2etest/connectivity_test.go

```diff
@@ -102,7 +102,7 @@ var (
 					Containers: []v1.Container{
 						{
 							Name:  "hello-tcp",
-							Image: "alpine/socat",
+							Image: "alpine/socat:1.7.4.4",
```

Comment:
It seems there's an upstream bug in socat causing this to fail: https://github.com/termux/termux-packages/issues/18645

This is the next most recent version, which doesn't seem to have the bug.

Reply:
should we use netcat instead? are the modern features in socat vs netcat actually being used in these connectivity tests?

Reply:
I don't have a strong opinion, but I was having a bit of trouble wiring up netcat to behave exactly like the UDP socat command. Even with `-k`, `nc -lkup 42 -e echo $MY_POD_ID` was exiting after the first request but the socat command stays alive. Using the same `nc` command without `-u` seems to work fine for TCP.